### PR TITLE
Options: avoid code duplication

### PR DESCRIPTION
**Pull request by smelc**
# Changelog

```yaml
- description: |
    Share code in parsers, by using cardano-api's [parseFilePath](https://github.com/IntersectMBO/cardano-api/blob/bf84187ae986f9a2621c55b690638fe8bb2cb156/cardano-api/internal/Cardano/Api/Utils.hs#L97) instead of duplicating its implementation in 50+ places.
# uncomment types applicable to the change:
  type:
  # - feature        # introduces a new feature
  # - breaking       # the API has changed in a breaking way
  # - compatible     # the API has changed but is non-breaking
  # - optimisation   # measurable performance improvements
  - refactoring    # QoL changes
  # - bugfix         # fixes a defect
  # - test           # fixes/modifies tests
  # - maintenance    # not directly related to the code
  # - release        # related to a new release preparation
  # - documentation  # change in code docs, haddocks...
```

# Context

Improvement found when starting to write https://github.com/input-output-hk/cardano-node-wiki/pull/52/files

# How to trust this PR

You can review the diff in Haskell files manually.

This question is more tricky for the 700+ golden files that have changed :smile: What I did is that I ran the golden tests against a modified version of cardano-api, at revision https://github.com/IntersectMBO/cardano-api/tree/bf84187ae986f9a2621c55b690638fe8bb2cb156. I changed this [line of parseFilePath](https://github.com/IntersectMBO/cardano-api/blob/bf84187ae986f9a2621c55b690638fe8bb2cb156/cardano-api/internal/Cardano/Api/Utils.hs#L101) so that it used `FILE` as the metavariable name. This way the diff is greatly reduced, because `FILE` was the existing metavariable used so far most of the time. I then reviewed the golden files changes manually and everything was good.

# Checklist

- [X] Commit sequence broadly makes sense and commits have useful messages
- [X] Self-reviewed the diff

### CODE REVIEW

File: cardano-cli/src/Cardano/CLI/EraBased/Options/Common.hs [hunk 1]
```diff
@@ -1574,12 +1437,10 @@ pTxBuildOutputOptions =
...
-      <$> Opt.strOption
-        ( Opt.long "calculate-plutus-script-cost"
-            <> Opt.metavar "FILE"
-            <> Opt.help "(File () Out) filepath of the script cost information."
-            <> Opt.completer (Opt.bashCompleter "file")
-        )
+      <$> File
+      <$> parseFilePath
+        "calculate-plutus-script-cost"
+        "(File () Out) filepath of the script cost information."
```
Comment (carbolymer) [inline cardano-cli/src/Cardano/CLI/EraBased/Options/Common.hs]: I know that it was copied from before, but I'm not sure about this haskell bit in the documentation: `(File () Out)`. Seems odd, and not really informative for people not familiar with cardano-cli codebase.

File: cardano-cli/src/Cardano/CLI/EraBased/Options/Common.hs [hunk 2]
```diff
@@ -288,14 +288,7 @@ pStakeVerificationKeyOrFile prefix =
...
 pScriptFor name Nothing help' =
-  fmap File $
-    Opt.strOption $
-      mconcat
-        [ Opt.long name
-        , Opt.metavar "FILE"
-        , Opt.help help'
-        , Opt.completer (Opt.bashCompleter "file")
-        ]
+  fmap File $ parseFilePath name help'
```
Comment (carbolymer) [inline cardano-cli/src/Cardano/CLI/EraBased/Options/Common.hs:291]: Nice, I didn't know we had this function. I guess we should move it from cardano-api to cardano-cli and don't depend on optparse-applicative in cardano-api.

**Review by carbolymer — APPROVED** [review]
Nice work! There are some HLint warnings though.

File: cardano-cli/src/Cardano/CLI/EraBased/Options/Common.hs (hunk 1 above)
Comment (smelc) [reply cardano-cli/src/Cardano/CLI/EraBased/Options/Common.hs]: Good point, I changed the message :+1:

File: cardano-cli/src/Cardano/CLI/EraBased/Options/Common.hs (hunk 2 above)
Comment (smelc) [reply cardano-cli/src/Cardano/CLI/EraBased/Options/Common.hs:291]: Good point, I'm looking whether we can do that easily :+1:

Comment (smelc) [reply cardano-cli/src/Cardano/CLI/EraBased/Options/Common.hs:291]: Follow-up PRs: https://github.com/IntersectMBO/cardano-api/pull/635 and https://github.com/IntersectMBO/cardano-cli/pull/894

**Pull request merged**